File: app/src/main/java/com/example/dkalev/remember/model/DeckDao.java
package com.example.dkalev.remember.model;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface DeckDao {

    @Query("SELECT * FROM deck")
    Flowable<List<Deck>> getAll();

    @Query("SELECT * FROM deck WHERE :deckId == deck_id")
    Flowable<Deck> getDeck(int deckId);

    @Insert
    void insertAll(Deck... decks);

    @Query("DELETE FROM deck WHERE :deckId == deck_id")
    void delete(int deckId);

    @Query("SELECT * FROM card WHERE :deckId == deck_id")
    Flowable<List<Card>> getCards(int deckId);
}

File: app/src/main/java/com/example/dkalev/remember/model/deck/DeckDao.kt
package com.example.dkalev.remember.model.deck

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import com.example.dkalev.remember.model.card.Card
import io.reactivex.Flowable
import java.util.*

@Dao
interface DeckDao {

    @get:Query("SELECT * FROM deck")
    val all: Flowable<List<Deck>>

    @Query("SELECT * FROM deck WHERE :deckName == deck_name")
    fun getDeck(deckName: String): Flowable<Deck>

    @Insert
    fun insertAll(vararg decks: Deck)

    @Query("DELETE FROM deck WHERE :deckName == deck_name")
    fun delete(deckName: String)

    @Query("SELECT * FROM card WHERE :deckName == deck_name")
    fun getCards(deckName: String): Flowable<List<Card>>

    @Query("SELECT * FROM card WHERE :deckName == deck_name AND :date >= due_date")
    fun getDueCards(deckName: String, date: Date): Flowable<List<Card>>
}